sdk/src/azure/core/az_json_token.c

```diff
@@ -333,7 +333,8 @@ AZ_NODISCARD az_result az_json_string_unescape(
 {
   _az_PRECONDITION_VALID_SPAN(json_string, 1, false);
   _az_PRECONDITION_NOT_NULL(destination);
-  _az_PRECONDITION(destination_max_size > 0);
```

Comment:
Interestingly, the precondition at line 335 precludes the more traditional use of this pattern:

```c
err = az_json_string_unescape(json_string, NULL, 0, &size);
char *dest = malloc(size);
err = az_json_string_unescape(json_string, dest, size, &size);
```

Instead, you need to do something silly like:
```c
char dummy;
err = az_json_string_unescape(json_string, &dummy, 0, &size);
char *dest = malloc(size);
err = az_json_string_unescape(json_string, dest, size, &size);
```

Reply:
Our existing SDK APIs don't behave like this, where the required size is returned on a failed call.

Reply:
That's a shame, because the entire reason for this pattern `([in]foo*, [in]int fooSize, [out]int* sizeNeeded)` is to enable that use scenario (passing in 0 and `NULL` to do a sizing pass followed by the actual allocation).